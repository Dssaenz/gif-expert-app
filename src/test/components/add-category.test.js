import React from 'react';
import AddCategory from '../../components/add-category';

import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

describe('Test in <AddCategory /> component', () => {
	const setCategories = jest.fn();

	let wrapper = shallow(<AddCategory setCategories={setCategories} />)

	beforeEach(()=> {
		jest.clearAllMocks();
		wrapper = shallow(<AddCategory setCategories={setCategories} />)
	})

	test('It should display the component correctly', () => {
		expect( wrapper ).toMatchSnapshot();
	})

	test('It should change the input', () => {
		const input = wrapper.find('input');
		const value = 'Hola mundo';

			input.simulate('change', { target: { value }});
			expect( wrapper.find('p').text().trim()).toBe(value);
	})

	test(`It shouldn't post the information with submit`, () => {
		const submit = wrapper.find('form');
		submit.simulate('submit', { preventDefault: () => {} })
		expect(setCategories).not.toHaveBeenCalled();
	})

	test(`Should call setCategories and have values in ''`, () => {
	const input = wrapper.find('input');
	const submit = wrapper.find('form');
	const value = 'hola mundo';

		input.simulate('change', { target: { value }});
		submit.simulate('submit', { preventDefault: setCategories })

		expect(setCategories).toHaveBeenCalled();
		expect(setCategories).toHaveBeenCalledTimes(1);
		// expect(setCategories).toHaveBeenCalledWith( expect.any(Function));

		expect(input.prop('value')).toBe('');

	})
})