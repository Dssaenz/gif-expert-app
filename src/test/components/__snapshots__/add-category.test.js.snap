// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AddCategory /> component It should display the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
